File: ToC_2026/8/question4/Java/3.java
import java.util.Scanner;

public class Q3_Lt {
    static boolean iszero(int m) { return m == 0; }
    static int pi(int m) { return m == 0 ? 0 : m - 1; }

    static boolean lt(int m, int n) {
        if (iszero(n)) {
            return false;
        } else if (iszero(m)) {
            return true;
        } else {
            return lt(pi(m), pi(n));
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter two numbers: ");
        int m = scanner.nextInt();
        int n = scanner.nextInt();
        if (lt(m, n)) {
            System.out.println(m + " is less than " + n + ".");
        } else {
            System.out.println(m + " is not less than " + n + ".");
        }
    }
}

File: ToC_2026/8/question4/C/3.c
//2つの自然数 m と n を取り，m < n の真理値を返す lt
#include <stdio.h>
int iszero(int m) { return m == 0; }
int pi(int m) { if (m == 0) return 0; return m - 1; }

int lt(int m, int n) {
    if (iszero(n)) {
        return 0; // nが先に0になったら m >= n
    } else if (iszero(m)) {
        return 1; // mが先に0になったら m < n
    } else {
        return lt(pi(m), pi(n));
    }
}

int main() {
    int m, n;
    printf("Enter two numbers: ");
    scanf("%d %d", &m, &n);
    if (lt(m, n)) {
        printf("%d is less than %d.\n", m, n);
    } else {
        printf("%d is not less than %d.\n", m, n);
    }
    return 0;
}